.docker/ci/Dockerfile

```diff
@@ -29,20 +32,55 @@ RUN \
         wget \
         # Required for rosdep command
         sudo \
-        # Preferred build tools
-        python-catkin-tools \
         clang clang-format-3.9 clang-tidy clang-tools \
-        ccache && \
+        ccache \
+        curl gnupg2 lsb-release \
+        # Some dependencies for working with ROS2
+        build-essential \
+        cmake \
+        git \
+        python3-colcon-common-extensions \
+        python3-pip \
+        python-rosdep \
+        python3-vcstool && \
+    #
+    # Upgrade Pip
+    python3 -m pip install --upgrade pip && \
+    #
+    # install some pip packages needed for testing
+    python3 -m pip install -U \
+        argcomplete \
+        flake8 \
+        flake8-blind-except \
+        flake8-builtins \
+        flake8-class-newline \
+        flake8-comprehensions \
+        flake8-deprecated \
+        flake8-docstrings \
+        flake8-import-order \
+        flake8-quotes \
+        git+https://github.com/lark-parser/lark.git@0.7d \
+        pytest-repeat \
+        pytest-rerunfailures \
+        pytest \
+        pytest-cov \
+        pytest-runner \
+        setuptools && \
+    #
+    # install Fast-RTPS dependencies
+    apt-get install -qq --no-install-recommends -y \
+        libasio-dev \
+        libtinyxml2-dev && \
     #
     # Download all dependencies of MoveIt!
     rosdep update && \
-    rosdep install -y --from-paths . --ignore-src --rosdistro ${ROS_DISTRO} --as-root=apt:false && \
+    rosdep install -y --from-paths . --ignore-src --rosdistro ${ROS_DISTRO} --as-root=apt:false --skip-keys "console_bridge fastcdr fastrtps libopensplice67 libopensplice69 rti-connext-dds-5.3.1 urdfdom_headers" && \
```

Comment:
Could you add a comment please, why these packages are skipped?

Reply:
These are skipped in the ros2 source install instructions so I merely copied them over. After testing, I don't believe we need to skip them

Reply:
for "hacky" looking code like this, please document where you got it from in the code (as you just did in the comments)

Reply:
You may want to break this over multiple indented lines to improve the traceability of changes in Dockerfiles via git blame etc. E.g. https://github.com/osrf/docker_images/blob/ea0fdabf250f816203d058dca1df12802a9c44cc/ros2/nightly/nightly/Dockerfile#L63-L72

Reply:
This is out of date. the current version is only a 101 characters wide